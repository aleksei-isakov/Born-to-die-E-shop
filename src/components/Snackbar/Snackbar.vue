<template>
  <div>
    <v-snackbar
      v-model="hasError"
      :timeout="timeout"
      top
      outlined
      vertical
      color="red"
    >
      {{ errorMessage }}

      <template v-slot:action="{ attrs }">
        <v-btn color="black" text v-bind="attrs" @click="closeGlobalMessage">
          Ok
        </v-btn>
      </template>
    </v-snackbar>
  </div>
</template>

<script>
import { mapGetters, mapActions } from 'vuex';

export default {
  name: 'Snackbar',

  data() {
    return {
      timeout: 2000
    };
  },

  computed: {
    ...mapGetters('globalmessagestate', ['isError', 'errorMessage']),

    hasError: {
      get() {
        return this.isError;
      },

      set() {
        this.closeGlobalMessage();
      }
    }
  },

  methods: {
    ...mapActions('globalmessagestate', ['closeGlobalMessage'])
  }
};
</script>
