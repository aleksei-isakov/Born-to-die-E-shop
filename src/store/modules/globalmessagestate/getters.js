const getters = {
  isError: (state) => state.isError,
  errorMessage: (state) => state.errorMessage
};

export default getters;
