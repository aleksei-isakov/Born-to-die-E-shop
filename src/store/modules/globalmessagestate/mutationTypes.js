export default {
  SHOW_MESSAGE: 'SHOW_MESSAGE',
  CLOSE_MESSAGE: 'CLOSE_MESSAGE'
};
