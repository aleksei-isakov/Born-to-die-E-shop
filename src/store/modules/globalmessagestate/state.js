const state = {
  isError: false,
  errorMessage: ''
};

export default state;
