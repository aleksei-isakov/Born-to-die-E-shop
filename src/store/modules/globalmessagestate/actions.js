import mutationTypes from './mutationTypes';

const actions = {
  showGlobalMessage({ commit }, payload) {
    commit(mutationTypes.SHOW_MESSAGE, payload);
  },
  closeGlobalMessage({ commit }) {
    commit(mutationTypes.CLOSE_MESSAGE);
  }
};

export default actions;
