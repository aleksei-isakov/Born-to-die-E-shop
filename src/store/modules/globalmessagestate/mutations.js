import mutationTypes from './mutationTypes';

const mutations = {
  [mutationTypes.SHOW_MESSAGE](state, payload) {
    state.isError = true;
    state.errorMessage = payload;
  },

  [mutationTypes.CLOSE_MESSAGE](state) {
    state.isError = false;
  }
};

export default mutations;
