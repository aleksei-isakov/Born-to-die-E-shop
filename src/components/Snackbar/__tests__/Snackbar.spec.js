import Vue from 'vue';
import Vuex from 'vuex';
import { createLocalVue, mount } from '@vue/test-utils';
import Snackbar from '../Snackbar';
import Vuetify from 'vuetify';
import globalmessagestate from '@/store/modules/globalmessagestate';

Vue.use(Vuetify);
const localVue = createLocalVue();

localVue.use(Vuex);

describe('Snackbar', () => {
  let store, vuetify, wrapper;
  const mockState = {
    errorMessage: "Hi, i'm error!"
  };

  beforeEach(() => {
    vuetify = new Vuetify();

    store = new Vuex.Store({
      modules: {
        globalmessagestate: {
          namespaced: true,
          state: mockState,
          getters: globalmessagestate.getters,
          actions: globalmessagestate.actions,
          mutations: globalmessagestate.mutations
        }
      }
    });
  });

  afterEach(() => {
    wrapper.destroy();
  });

  test('component should render correctly', () => {
    wrapper = mount(Snackbar, {
      store,
      localVue,
      vuetify
    });
    expect(mockState.errorMessage).toEqual("Hi, i'm error!"),
      expect(wrapper).toMatchSnapshot();
  });
});
