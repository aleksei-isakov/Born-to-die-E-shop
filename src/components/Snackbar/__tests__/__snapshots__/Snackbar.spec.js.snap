// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snackbar component should render correctly 1`] = `
<div class="v-snack v-snack--top v-snack--vertical">
  <transition-stub name="v-snack-transition">
    <div class="v-snack__wrapper v-sheet v-sheet--outlined theme--light red--text" style="display: none;">
      <div role="status" aria-live="polite" class="v-snack__content">
        Hi, i'm error!

      </div>
      <div class="v-snack__action "><button type="button" class="v-btn v-btn--text theme--light v-size--default black--text v-snack__btn"><span class="v-btn__content">
      Ok
    </span></button></div>
    </div>
  </transition-stub>
</div>
`;
